server/src/main/java/io/spine/server/BoundedContext.java

```diff
@@ -539,4 +458,117 @@ public String toString() {
+            self().register(checkNotNull(repository));
+        }
+
+        /**
+         * Registers the passed command dispatcher.
+         *
+         * @see BoundedContext#registerCommandDispatcher(CommandDispatcher)
+         */
+        public void registerCommandDispatcher(CommandDispatcher dispatcher) {
+            self().registerCommandDispatcher(checkNotNull(dispatcher));
+        }
+
+        /**
+         * Registers the passed event dispatcher.
+         *
+         * @see BoundedContext#registerEventDispatcher(EventDispatcher)
+         */
+        public void registerEventDispatcher(EventDispatcher dispatcher) {
+            self().registerEventDispatcher(dispatcher);
+        }
+
+        /**
+         * Obtains repositories of the context.
+         *
+         * @throws IllegalStateException
+         *         if there is not repository entities of which have the passed state
+         */
+        public Repository<?, ?> getRepository(Class<? extends EntityState> stateClass) {
+            return guard.get(stateClass);
+        }
+
+        /**
+         * Finds a repository by the state class of entities.
+         *
+         * <p>This method assumes that a repository for the given entity state class <b>is</b>
+         * registered in this context. If there is no such repository, throws
+         * an {@link IllegalStateException}.
+         *
+         * <p>If a repository is registered, the method returns it or {@link Optional#empty()} if
+         * the requested entity is {@linkplain Visibility#NONE not visible}.
+         *
+         * @param stateClass
+         *         the class of the state of the entity managed by the resulting repository
+         * @return the requested repository or {@link Optional#empty()} if the repository manages
+         *         a {@linkplain Visibility#NONE non-visible} entity
+         * @throws IllegalStateException
+         *         if the requested repository is not registered
+         * @see VisibilityGuard
+         */
+        public Optional<Repository<?, ?>> findRepository(Class<? extends EntityState> stateClass) {
+            // See if there is a repository for this state at all.
+            if (!guard.hasRepository(stateClass)) {
+                throw newIllegalStateException(
+                        "No repository found for the entity state class `%s`.",
+                        stateClass.getName()
+                );
+            }
+            Optional<Repository<?, ?>> repository = guard.repositoryFor(stateClass);
+            return repository;
+        }
+
+        /** Obtains an {@link IntegrationBroker} of this {@code BoundedContext}. */
+        public IntegrationBroker broker() {
+            return self().broker;
+        }
+
+        /**
+         * Obtains a tenant index of this Bounded Context.
+         *
+         * <p>If the context is single-tenant returns
+         * {@linkplain TenantIndex#singleTenant() null-object} implementation.
```

Comment:
I think, it's not a null-object per se. Just like an `ImmutableList.of(singleElement)` is not a null-object.

Reply:
Single tenant means there is no tenancy in this context. And it does not contains a reference to the _real_ tenant ID value. In this case, it's just a constant, which says “there are no tenants here”. That's why it's a null object.